ggshield/cmd/secret/scan/prereceive.py

```diff
@@ -155,19 +164,21 @@ def prereceive_cmd(
         )
         return 0
 
-    before, after = parse_stdin()
+    before_after = parse_stdin()
+    if before_after is None:
+        click.echo("Deletion event or nothing to scan.", err=True)
+        return 0
+
+    before, after = before_after
     if before == after:
         click.echo(
             "Pushed branch does not contain any new commit.",
             err=True,
         )
         return 0
 
-    if after == EMPTY_SHA:
-        click.echo("Deletion event or nothing to scan.", err=True)
-        return 0
-
     assert before != EMPTY_SHA
+    assert after != EMPTY_SHA
```

Comment:
We already handled the `if before_after is None`, do we need this ?

Reply:
Since I removed the `if after == EMPTY_SHA:`, I feel more confident that case is indeed handled by the `if before_after is None` with this asset in place.

Reply:
Okay, that's not a blocker for me of course. 
But if we take it the other way around : 
- if this assert fails, it means after == EMPTY_SHA
- This means in turn that before_after is something like (_, EMPTY_SHA)
- This means that parse_stdin() returned such a tuple
- Looking at the code for parse_stdin, this means that `new_commit` was EMPTY_SHA
- But if new_commit is EMPTY_SHA, parse_stdin returns None.
That's why the assert looks useless to me.

Reply:
Your analysis is correct, but the goal of this assert is not to get triggered by the current code. It has two jobs:

1. Document the assumption made while writing this code (useful for future code readers)
2. Trigger if we do something wrong in the future, like modifying `parse_stdin()` and making it return `EMPTY_SHA`, invalidating our assumption

It is a second line of defense. An assert triggering indicates a programmer error, it must never trigger for normal program behavior (including user errors).

Reply:
Super clear, thanks for the explanation.